src/Compilers/Core/Portable/CaseInsensitiveComparison.cs

```diff
@@ -300,7 +303,8 @@ public static int GetHashCode(string value)
         /// </summary>
         /// <param name="value"></param>
         /// <returns></returns>
-        public static string ToLower(string value)
+        [return: NotNullIfNotNull(parameterName: "value")]
+        public static string? ToLower(string value)
```

Comment:
💭 It's strange to see `NotNullIfNotNull` without the parameter being nullable.

```suggestion
        public static string? ToLower(string? value)
```